src/goLanguageServer.ts

```diff
@@ -311,39 +298,43 @@ export function getLanguageServerToolPath(): string {
 		return;
 	}
 
-	// Get the path to gopls or any alternative that the user might have set for gopls.
-	const goplsBinaryPath = getBinPath('gopls');
-	if (path.isAbsolute(goplsBinaryPath)) {
-		return goplsBinaryPath;
-	}
-
-	// Get the path to go-langserver or any alternative that the user might have set for go-langserver.
-	const golangserverBinaryPath = getBinPath('go-langserver');
-	if (path.isAbsolute(golangserverBinaryPath)) {
-		return golangserverBinaryPath;
-	}
-
-	// If no language server path has been found, notify the user.
+	// Determine which language server the user has selected.
+	// gopls is the default choice.
 	let languageServerOfChoice = 'gopls';
 	if (goConfig['alternateTools']) {
 		const goplsAlternate = goConfig['alternateTools']['gopls'];
+
+		// Check if the user has set the deprecated "go-langserver" setting.
 		const golangserverAlternate = goConfig['alternateTools']['go-langserver'];
-		if (typeof goplsAlternate === 'string') {
+		if (golangserverAlternate) {
+			vscode.window.showErrorMessage(`The "go.alternateTools" setting for "go-langserver" has been deprecated. Please use "gopls" instead.`);
```

Comment:
Do you mean "language-server", or are you suggesting "go-langserver" is not supported at all and switch to use "gopls"? 

And, the user needs to reload the window, right? Mention that in the error message.

Reply:
Mentioned reloading in the error message.

I realized that the `"go-langserver"` setting was actually not supported in the `package.json` at all, so I figured I'd first start by fully removing it in the code. In a follow-up, I will change the setting to be `"language-server"`, if @ramya-rao-a is okay with that.